RxJavaTemplate/app/src/main/java/co/nimblehq/rxjava/extension/SharePreferencesExt.kt

```diff
@@ -0,0 +1,22 @@
+package co.nimblehq.rxjava.extension
+
+import android.content.SharedPreferences
+
+fun SharedPreferences.execute(operation: (SharedPreferences.Editor) -> Unit) {
+    with(edit()) {
+        operation(this)
+        apply()
+    }
+}
+
+fun SharedPreferences.Editor.put(pair: Pair<String, Any?>) {
+    val key = pair.first
+    when (val value = pair.second) {
+        is String -> putString(key, value)
+        is Int -> putInt(key, value)
+        is Boolean -> putBoolean(key, value)
+        is Long -> putLong(key, value)
+        is Float -> putFloat(key, value)
+        else -> error("Only primitive types can be stored in SharedPreferences")
+    }
+}
```

Comment:
I'd suggest having the inline function with generic type inside the BaseSharedPreferences like this

```
    protected inline fun <reified T> get(key: String): T? =
        if (prefs.contains(key)) {
            when (T::class) {
                Boolean::class -> prefs.getBoolean(key, false) as T?
                String::class -> prefs.getString(key, null) as T?
                Float::class -> prefs.getFloat(key, 0f) as T?
                Int::class -> prefs.getInt(key, 0) as T?
                Long::class -> prefs.getLong(key, 0L) as T?
                else -> null
            }
        } else {
            null
        }
```

```
    protected fun <T> set(key: String, value: T?) {
        prefs.edit().apply {
            when (value) {
                null -> remove(key)
                is Boolean -> putBoolean(key, value)
                is String -> putString(key, value)
                is Float -> putFloat(key, value)
                is Long -> putLong(key, value)
                is Int -> putInt(key, value)
            }
        }.apply()
    }
```

Then from inherited classes:

```
    private var userId: Int?
        get() = get(PREFS_KEY_USER_ID)
        set(value) = set(PREFS_KEY_USER_ID, value)
```

It would look more simple than having getter/setter methods from SharedPreferences classes